src/swap_amm.py

```diff
@@ -324,25 +325,21 @@ def __init__(self, symbol1, symbol2, addresses_dict, myswap_id):
     async def get_balance(self):
         if self.myswap_id is not None:
             myswap_pool = await src.blockchain_call.get_myswap_pool(self.myswap_id)
-        for amm, addresses in self.addresses_dict.items():
+        for amm, address in self.addresses_dict.items():
             for token in self.tokens:
-                balance = 0
-                if isinstance(addresses, str):
-                    addresses = [addresses]
-                for address in addresses:
-                    balance += await src.blockchain_call.balance_of(token.address, address)
-                if self.myswap_id is not None:
+                balance = await src.blockchain_call.balance_of(token.address, address)
+                if self.myswap_id is not None and amm == "MySwap":
                     balance += myswap_pool[token.symbol.upper()]
```

Comment:
Now MySwap is still added twice, right? First on line 330, and then on line 332.